src/common/snap_types.cc

```diff
@@ -20,6 +21,9 @@ void SnapRealmInfo::decode(ceph::buffer::list::const_iterator& bl)
   decode(h, bl);
   ceph::decode_nohead(h.num_snaps, my_snaps, bl);
   ceph::decode_nohead(h.num_prior_parent_snaps, prior_parent_snaps, bl);
+  if (!bl.end()) {
```

Comment:
This is a bit off-putting. I think we should make this versioned. @lxbsz

Reply:
Yeah, that's better. Or in future it's will be hard to add new members.

Reply:
Will send a separate PR for this.